tests/unit/lms/db/_bulk_action_test.py

```diff
@@ -35,16 +34,17 @@ def test_upsert(self, db_session):
         )
         db_session.flush()
 
-        result = BulkAction(db_session).upsert(
-            self.TableWithBulkUpsert,
-            [
-                {"id": 1, "name": "update_old", "other": "post_1"},
-                {"id": 3, "name": "create_with_id", "other": "post_3"},
-                {"id": 4, "name": "over_block_size", "other": "post_4"},
-            ],
-        )
+        upsert_rows = [
```

Comment:
I didn't move these tests, just tweaked them to make sure that the changes didn't break it.